src/utils/equipment.ts

```diff
@@ -0,0 +1,87 @@
+import Equipment from "../resources/Equipment";
+// Filtering Function to reduce the size of the equipment array being passed down
+export function queryEquipment(
+  equipment: Equipment[],
+  searchString: string
+): Equipment[] {
+  let queriedEquipment: Equipment[] = [];
+  if (searchString !== "") {
+    const queries = searchString.split(/\W+/);
+    queriedEquipment = equipment.filter(function (equipment) {
+      return queries.every(function (query) {
+        return (
+          equipment.description
+            .toLowerCase()
+            .includes(query.toLowerCase().trim()) ||
+          equipment.category
+            .toLowerCase()
+            .includes(query.toLowerCase().trim()) ||
+          equipment.tags.some(function (tag) {
+            return tag.name.toLowerCase().includes(query.toLowerCase().trim());
+          })
+        );
+      });
+    });
+  } else {
+    queriedEquipment = equipment;
+  }
+  return queriedEquipment;
+}
+
+// Filtering Function to reduce the size of the equipment array being passed down
+export function filterEquipment(
+  equipment: Equipment[],
+  filters: { [k: string]: boolean }
+): Equipment[] | undefined {
+  const activeFilters = Object.keys(filters).filter(function (key: string) {
+    return filters[key];
+  });
+  return equipment.filter(function (item) {
+    return activeFilters.every(function (filter) {
+      return item.tags.some(function (tag) {
+        return tag.name.toLowerCase().includes(filter.toLowerCase().trim());
+      });
+    });
+  });
+}
+
+export function quantizeEquipment(equipment: Equipment[]): Equipment[] {
+  return equipment.reduce((quantized: Equipment[], item) => {
+    const hasTitle = (equipment: Equipment): boolean =>
+      equipment.description === item.description;
+    const alreadyDone = quantized.find(hasTitle);
+    if (!alreadyDone)
+      quantized.push({
+        ...item,
+        quantity: equipment.filter(hasTitle).length,
+      });
+    return quantized;
+  }, []);
+}
+
+// Function to convert equipment Array to Quantized equipment Array
+export function buildDictionaries(
+  equipment: Equipment[]
+): [
+  { [k: string]: boolean },
+  { [k: string]: Set<string> },
+  { [k: string]: number }
+] {
+  // Build quantities dictionary for Formik
+  // Build Categories Dictionary
+  // Build Filters Dictionary
+  const filters: { [k: string]: boolean } = {};
+  const categories: { [k: string]: Set<string> } = {};
+  const quantities: { [k: string]: number } = {};
+  equipment.forEach((item) => {
+    quantities[item.description] = 0;
+    item.tags.forEach((tag) => {
+      if (!categories[item.category]) {
+        categories[item.category] = new Set();
+      }
+      categories[item.category].add(tag.name);
+      filters[tag.name] = false;
+    });
+  });
+  return [filters, categories, quantities];
+}
```

Comment:
This is basically the same (double check that it runs the same...) but replacing the outer `.forEach` with a `.reduce`.
I'm not 100% on this (including the suggestion).  The inner `.forEach` has a bit of a [smell](https://en.wikipedia.org/wiki/Code_smell) to it... perhaps this function is doing to much.  It is hard to reason about. I haven't looked at where it's called yet.
```suggestion
// Build quantities dictionary for Formik
// Build Categories Dictionary
// Build Filters Dictionary
export function buildDictionaries(
  equipment: Equipment[]
): [
  { [k: string]: boolean },
  { [k: string]: Set<string> },
  { [k: string]: number }
] {
  return equipment.reduce(
    ([filters, categories, quantities], item) => {
      quantities[item.description] = 0;
      item.tags.forEach((tag) => {
        if (!categories[item.category.name]) {
          categories[item.category.name] = new Set();
        }
        categories[item.category.name].add(tag.name);
        filters[tag.name] = false;
      });
      return [filters, categories, quantities];
    },
    [
      {} as { [k: string]: boolean },
      {} as { [k: string]: Set<string> },
      {} as { [k: string]: number },
    ]
  );
}
```